new-packages/ts-project/__tests__/source-edits/add-transition.test.ts

```diff
@@ -0,0 +1,1830 @@
+import { expect, test } from 'vitest';
+import { createTestProject, testdir, ts } from '../utils';
+
+test('should be possible to add a targetless transition', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { creteMachine } from "xstate";
+
+      createMachine({});
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+
+  const textEdits = project.editDigraph(
+    {
+      fileName: 'index.ts',
+      machineIndex: 0,
+    },
+    {
+      type: 'add_transition',
+      sourcePath: [],
+      targetPath: null,
+      transitionPath: ['on', 'FOO', 0],
+    },
+  );
+  expect(await project.applyTextEdits(textEdits)).toMatchInlineSnapshot(`
+    {
+      "index.ts": "import { creteMachine } from "xstate";
```

Comment:
`crete` -> `create`

Reply:
aaaaargh, i keep copy-pasting the same typo 😂